src/kudu/consensus/raft_consensus.cc

```diff
@@ -3097,6 +3097,18 @@ void RaftConsensus::DisableFailureDetector() {
   }
 }
 
+void RaftConsensus::SetWitholdVotesForTests(bool withold_votes) {
+  if (withold_votes) {
+    withhold_votes_until_ = MonoTime::Max();
+  } else {
+    withhold_votes_until_ = MonoTime::Min();
+  }
+}
+
+void RaftConsensus::SetRejectAppendEntriesForTests(bool reject_append_entries) {
+  FLAGS_follower_reject_update_consensus_requests = reject_append_entries;
```

Comment:
this wont work in unit tests or mock tests, because all raft consensus objects will see the flags change.

Reply:
Not sure what you mean. This is false by default. I will set it to true using a server variable (D21990305). Let me know if you still think it won't work.

Reply:
As per the conversation on messenger. Merlin tests or any tests running on separate processes per instance would NOT be affected. I have changed the code to use a bool instead of a gflag so that if in the future there are tests with multiple instances of RaftConsensus within the same process, they could also utilize this.